### PR TITLE
ci: Build wheels in separate workflow

### PR DESCRIPTION
**Pull request by j-s-ashley**
Addresses part of #31:

> - [ ] Split the CI components from the wheel building in CI
> c.f. https://github.com/thaler-lab/Wasserstein/pull/8
> - [ ] Have the wheel building workflows run and [upload the wheels to GitHub Actions as an artifact](https://learn.scientific-python.org/development/guides/gha-wheels/#the-core-job-3-main-oss)

### CODE REVIEW

File: .github/workflows/build-wheels.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository and submodules
+        uses: actions/checkout@v4
+        with:
+          submodules: recursive
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
```
Comment (matthewfeickert) [inline .github/workflows/build-wheels.yml:29]:
```suggestion
        uses: actions/setup-python@v5
```

Not your fault, but you've been patiently waiting for review so long that there's a new version out. :sweat_smile:

File: .github/workflows/build-wheels.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        run: pipx run build
+
+      - name: Upload to GitHub
+        uses: actions/upload-artifact@v4
+        with:
+          name: Packages
+          path: dist/*
+
+      - name: Check metadata
+        run: pipx run twine check dist/*
```
Comment (matthewfeickert) [inline .github/workflows/build-wheels.yml:44]:
```suggestion
        run: pipx run twine check --strict dist/*
```

`twine` has a strict option.

```console
$ pipx run twine check --help
usage: twine check [-h] [--strict] dist [dist ...]

positional arguments:
  dist        The distribution files to check, usually dist/*

options:
  -h, --help  show this help message and exit
  --strict    Fail on warnings
```

**Review by matthewfeickert — CHANGES_REQUESTED** [review]
Looks good beyond some very small changes.

Totally optional but you can also add some visual checks that will show up in the logs (probably post `twine check`) with

```yaml
    - name: List contents of sdist
      run: python -m tarfile --list dist/*.tar.gz

    - name: List contents of wheel
      run: python -m zipfile --list dist/*.whl
```

**Review by matthewfeickert — COMMENTED** [review]
LGTM. Just 1 nitpick suggestion.

File: .github/workflows/build-wheels.yml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+          path: dist/*
+
+      - name: Check metadata
+        run: pipx run twine check --strict dist/*
+
+      - name: List contents of sdist
+        run: python -m tarfile --list dist/*.tar.gz
+
+      - name: List contents of wheel
+        run: python -m zipfile --list dist/*.whl
```
[the remark this answers is not in the document]
Comment (j-s-ashley) [reply .github/workflows/build-wheels.yml:50]: Gotcha. No, that makes a lot of sense. Thanks!

**Review by matthewfeickert — APPROVED** [review]

**Pull request merged**